apps/backend/src/libs/modules/udemy/base-udemy.module.ts

```diff
@@ -0,0 +1,105 @@
+import { HTTPOptions } from "shared";
+
+import { HTTPHeader, Http } from "../http/http.js";
+import { ApiPath } from "./libs/enums/enums.js";
+import { SearchParameters, Udemy } from "./libs/types/udemy.type.js";
+
+type Constructor = {
+	baseUrl: string;
+	clientId: string;
+	clientSecret: string;
+	fields: {
+		course: Record<string, string>;
+		courseCurriculum: Record<string, string>;
+		courseDetails: Record<string, string>;
+	};
+	http: Http;
+};
+
+class BasseUdemy<C, CD, CC> implements Udemy<C, CD, CC> {
```

Comment:
```suggestion
class Udemy {
```
you do not need a type for implementation

just describe all methods that you need and use this class a type to pass to other services